tests/rados/test_scrub_enhancement.py

```diff
@@ -0,0 +1,204 @@
+"""
+Program is for testing the scrub enhancement feature.As part of feature testing creating
+corrupted snapshot object.Performing scrub/deep-scrub to remove the corrupted SNA_ objects and
+replicating the correct SNA_ object
+"""
+
+import random
+import time
+import traceback
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
+from tests.rados.rados_test_util import create_pools
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Verification of the scrub enhancement feature
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+    try:
+        log.info(run.__doc__)
+        global rados_obj, acting_osd_node, config
+        config = kw["config"]
+        cephadm = CephAdmin(cluster=ceph_cluster, **config)
+        rados_obj = RadosOrchestrator(node=cephadm)
+        pool_obj = PoolFunctions(node=cephadm)
+        client_node = ceph_cluster.get_nodes(role="client")[0]
+        mount_point = "~/automation_dir"
+        test_scenarios = ["scrub", "deep-scrub"]
+        log.info("Running scrub enhancement feature testing")
+
+        for test_input in test_scenarios:
+            log.info(f"The feature is testing withe the {test_input}")
+            log.info("Creating the pool with the single PG")
+            create_pools(config, rados_obj, client_node)
+            log.info("Pool creation completed")
+            oname = "UNIQUEOBJECT{i}".format(i=random.randint(0, 10000))
+            pool_name = config["create_pools"][0]["create_pool"]["pool_name"]
+            log.info("Setting the autoscale mode off")
+            rados_obj.set_pool_property(
+                pool=pool_name, props="pg_autoscale_mode", value="off"
+            )
+            log.info(f"The {pool_name} autoscale mode is off")
+            log.info(f"creating the object inside the {pool_name}")
+            pool_obj.do_rados_put(client=client_node, pool=pool_name, obj_name=oname)
+            log.info(f"A object created inside the {pool_name}")
+            snapcmd = f"sudo rados mksnap -p {pool_name} snap1"
+            client_node.exec_command(cmd=snapcmd, sudo=True)
+            log.info("Creating the SNA_ entry by creating object")
+            pool_obj.do_rados_put(client=client_node, pool=pool_name, obj_name=oname)
```

Comment:
We're pulling the same object in line 52, and again here in line 57. 
Also, since the object name is same in both the PUT's, would it not be overwritten ?

How can same object name have SNA_ entry during 2nd put?

Reply:
It is a valid question. Can you raise a bug?

Reply:
I will raise the Bug and provide the details

Reply:
Please add the bugzilla details in the code as comments, so that it can be referenced and code can be fixed when bug is resolved.